predicators/train_refinement_estimator.py

```diff
@@ -0,0 +1,285 @@
+
+To specify a dataset file name different from the default:
+    python predicators/train_refinement_estimator.py --env narrow_passage \
+        --approach refinement_estimation --refinement_estimator tabular \
+        --seed 0 --load_data --data_file_name my_data_file.data
+
+To skip training the approach:
+    python predicators/train_refinement_estimator.py --env narrow_passage \
+        --approach refinement_estimation --refinement_estimator tabular \
+        --seed 0 --skip_training
+"""
+
+import logging
+import os
+import sys
+import time
+from collections import defaultdict
+from pathlib import Path
+from typing import List, Set, cast
+
+import dill as pkl
+
+from predicators import utils
+from predicators.approaches import create_approach
+from predicators.approaches.refinement_estimation_approach import \
+    RefinementEstimationApproach
+from predicators.envs import BaseEnv, create_new_env
+from predicators.ground_truth_nsrts import get_gt_nsrts
+from predicators.option_model import _OptionModelBase, create_option_model
+from predicators.planning import PlanningFailure, PlanningTimeout, \
+    _MaxSkeletonsFailure, _skeleton_generator, _SkeletonSearchTimeout, \
+    run_low_level_search
+from predicators.settings import CFG
+from predicators.structs import NSRT, Metrics, ParameterizedOption, \
+    Predicate, RefinementDatapoint, Task, Type
+
+assert os.environ.get("PYTHONHASHSEED") == "0", \
+        "Please add `export PYTHONHASHSEED=0` to your bash profile!"
+
+
+def train_refinement_estimation_approach() -> None:
+    """Main function, setting up env/approach and then performing data
+    collection and/or training as required."""
+    # Following is copied/adapted from predicators/main.py
+    script_start = time.perf_counter()
+
+    # Parse & validate args
+    args = utils.parse_args()
+    utils.update_config(args)
+    str_args = " ".join(sys.argv)
+
+    # Log to stderr.
+    handlers: List[logging.Handler] = [logging.StreamHandler()]
+    if CFG.log_file:
+        handlers.append(logging.FileHandler(CFG.log_file, mode='w'))
+    logging.basicConfig(level=CFG.loglevel,
+                        format="%(message)s",
+                        handlers=handlers)
+    if CFG.log_file:
+        logging.info(f"Logging to {CFG.log_file}")
+    logging.info(f"Running command: python {str_args}")
+
+    # Create environment
+    env = create_new_env(CFG.env, do_cache=True, use_gui=CFG.use_gui)
+    # The action space and options need to be seeded externally, because
+    # env.action_space and env.options are often created during env __init__().
+    env.action_space.seed(CFG.seed)
+    for option in env.options:
+        option.params_space.seed(CFG.seed)
+    assert env.goal_predicates.issubset(env.predicates)
+    preds, _ = utils.parse_config_excluded_predicates(env)
+
+    # Create the train tasks.
+    train_tasks = env.get_train_tasks()
+    # If train tasks have goals that involve excluded predicates, strip those
```

Comment:
as mentioned on slack, let's get rid of this here, which will resolve the pickling issue